### ISSUE TITLE
Support Faraday 2.x

### ISSUE THREAD

**Issue by tmimura39** [zendesk/zendesk_api_client_rb#488]
[Faraday 2.x has been released.](https://github.com/lostisland/faraday/blob/fc112254746ae92e3dcc8e6f9055878b1c583cc7/CHANGELOG.md#200-2022-01-04)

but, faraday is require to `< 2.0.0` in the gemspec of zendesk_api.
https://github.com/zendesk/zendesk_api_client_rb/blob/8665b8e9bfb638805be550ab54aca0ac59b86271/zendesk_api.gemspec#L29

Is it possible to relax the dependency?

**Comment by jrmhaig** [comment]
There are breaking changes so simply relaxing the dependency is not enough.

* The `dependency` has been removed. This is used in `lib/zendesk_api/middleware/request/encode_json.rb`.
* `Faraday::Response::Middleware` has been removed. This is used in various places.
* Possibly various other changes too. See the [upgrading guide.](https://github.com/lostisland/faraday/blob/main/UPGRADING.md)

**Comment by rsl** [comment]
so is the plan to just never upgrade? that really hurts those developers who want to keep their software up-to-date. that reply seems a little "not our problem/not planning to address it/old versions work for us".

**Comment by jrmhaig** [comment]
@rsl I absolutely agree and yes, my reply was not an official reply. I was hoping that I had provided enough information so that the Zendesk developers could work on it and am a bit disappointed to not even have received an acknowledgement. I did start creating a fix myself but when I saw that it was more than a few minutes, and we are paying Zendesk for their service, I would really want paying for my time!

We are using Faraday elsewhere in our app and I would like to upgrade those parts to version 2 so this issue is a blocker for me.

**Comment by rsl** [comment]
yeh sadly, we're/i'm looking at ditching this gem [i've ran into other issues besides this] and making our own light wrapper around their REST API. i don't expect them to look into this. i forked the gem and looked at making the changes to make a PR [like a good OSS citizen] but man, those changes are VERY out of my league in terms of knowing how to change out the middleware approaches and things.

i couldn't even get the specs to run out of the box with that json change about `dependency`.

**Comment by ecoologic** [comment]
Hi there,

This update is not a trivial change, but we are likely to pick this up soon. We could surely have done a better job at keeping you informed and we hope to do a better job from now on.

**Comment by rsl** [comment]
oh yeah. i tried looking into it [i've contributed to other OSS projects] and unfortunately the changes in the Faraday internals [as well as just lower-level HTTP in general] were definitely daunting and caused me to back away.

**Comment by rsl** [comment]
that PR is looking good. just wanted to give some encouragement.

**Comment by ecoologic** [comment]
🎉 Good news! There's a release candidate version now available `v1.38.rc1` that supports Faraday 2.

Please take a look and give us feedback, we'll do our best to be present and correct any problem with it. ❤️

**Comment by LandonSchropp** [comment]
@ecoologic Any chance there'll be a full release for this sometime soon?

**Comment by emmawertz-drizly** [comment]
Hi! Installing `zendesk_api` version `1.38.0.rc1` or newer gives me this error:
```
zendesk_api (~> 1.38.0.rc1) was resolved to 1.38.0.rc2, which depends on
      faraday (> 2.0.0)
```

But my `faraday` version is `2.1.0`...
